### PR TITLE
snap,snap-exec: support command-chain for hooks

### PR DESCRIPTION
**Pull request by kyrofa**
When a snap is created by the snapcraft CLI, each hook potentially has an associated chunk of environment variables defined as well as scripts that may be run before the hook itself. Currently, this is done by generating a shell script wrapper for each hook, but this is opaque and confusing.

Snapd supports an `environment` keyword for apps and hooks as well as a `command-chain` for apps. Add the last item needed before the shell wrappers can go away completely: `command-chain` support in hooks.

See forum topic https://forum.snapcraft.io/t/6112 for more details.

### CODE REVIEW

File: tests/main/command-chain/task.yaml [hunk 1]
```diff
@@ -9,7 +17,12 @@ restore: |
     rm -f command-chain_1.0_all.snap
 
 execute: |
-    echo "Test that command-chain actually runs as expected"
+    echo "Test that command-chain runs for hooks"
+    [ "$(cat "$BREADCRUMB")" = "chain1 chain2 configure" ]
+    cat "$ENVDUMP" | MATCH '^CHAIN_1_RAN=1$'
```
Comment (bboozzoo) [inline tests/main/command-chain/task.yaml]:
```
MATCH '^CHAIN_1_RAN=1$' < "$ENVDUMP"
```

Comment (kyrofa) [reply tests/main/command-chain/task.yaml]: Sure, updated.

**Comment by kyrofa** [comment]
Ping @bboozzoo, @chipaca.

**Comment by kyrofa** [comment]
> We should probably tune a bit how the skipping logic takes place, but that can be an independent PR.

[Here](https://github.com/snapcore/snapd/pull/5723) is the followup.

**Pull request merged**